Features/login api

mohamedsyoussef opened the pull request:
https://github.com/mohamedsyoussef/doctor_appointment/assets/72705926/9409e1f0-0fbd-40e0-b2ae-3686b8d7c978

mohamedsyoussef committed:
Add Android 12 splash screen configurations and assets

Added Android 12 splash screen configurations and assets for different screen densities and modes. Updated styles for splash screen background color and animated icon.

mohamedsyoussef committed:
Add new text styles and form field widget

Added new text styles for different font sizes and weights. Also, introduced a new
custom text form field widget with customizable properties like content padding,
borders, text styles, and suffix icons.

mohamedsyoussef committed:
Update AddTextFormField class to use isObscureText instead of obscureText

Changed the property name from obscureText to isObscureText in the
AddTextFormField class to improve clarity and consistency with its usage
in the LoginScreen widget. This change ensures better readability and
understanding of the code.

mohamedsyoussef committed:
Update AddTextFormField constructor parameters

Add a new parameter suffixIconColor to the AddTextFormField constructor
and set the default color for the suffix icon to ColorManager.mainBlue.
Adjust the border width in the TextFormField styling. Fix email field in
LoginScreen to not obscure text.

mohamedsyoussef committed:
Add ColorManager class for managing colors and update color imports

- Add ColorManager class to manage colors in the app
- Update color imports in various files to use ColorManager instead of colors file

mohamedsyoussef committed:
Update text styles and button appearance

Add a new text style for a 12px blue bold font and update button appearance
to use the main blue color and a larger border radius.

mohamedsyoussef committed:
Add Cupertino import and adjust text styles in login screen

Import Cupertino package for future use and adjust text styles in the
login screen for better readability and consistency.

mohamedsyoussef committed:
Add build_runner and json_serializable dependencies

Add build_runner and json_serializable dependencies to dev_dependencies in pubspec.yaml to enable Dart code generation and modular compilation.

mohamedsyoussef committed:
Add ApiErrorModel class for handling API errors

Add ApiErrorModel class with toJson and fromJson methods for handling API errors. Also, include generated part file for serialization. Update build_runner version to 2.4.8.

mohamedsyoussef committed:
Add new networking files and login request body model

Added new networking files for API constants and service along with a login request body model for the login feature.

mohamedsyoussef committed:
Add login API service method and related models

Added a new login API service method in ApiService class to handle login requests
and implemented related models LoginResponse and UserData for response parsing.

mohamedsyoussef committed:
Add dependency injection setup and login feature logic

- Add dependency injection setup using GetIt
- Implement login feature logic with LoginRepo and LoginCubit
- Update main.dart to include dependency injection setup
- Add flutter_bloc and provider dependencies
- Update pubspec.yaml with new dependencies

mohamedsyoussef committed:
Refactor file imports and add onboarding screen components

- Refactored file imports in app_router.dart and login_cubit.dart
- Added OnBoardingScreen, DoctorImage, and DocLogo components for onboarding screen

mohamedsyoussef committed:
Add password validation widgets and update login screen UI

- Added password validation widgets for lowercase, uppercase, special
  characters, numbers, and minimum length.
- Updated the login screen UI to include these password validation
  widgets and refactored the code for better structure and readability.

mohamedsyoussef committed:
Refactor home screen UI layout

Update home screen UI layout to center text and increase font size for better
visibility.

aboelnasrmo reviewed (changes requested):
Try to prevent to go back to on boarding screen, with the current code if the user go back , he will get error for the text editing controller was used after being disposed 
The screen should be Signup not sign in , in the login screen no need to check the regex

aboelnasrmo commented on lib/features/login/ui/widgets/email_password.dart:
```diff
@@ -0,0 +1,104 @@
+            suffixIcon: const Icon(Icons.email),
+            hintText: 'Enter Your E-mail',
+            validator: (value) {
+              if (value == null ||
```
can return null

aboelnasrmo commented on lib/features/login/ui/widgets/email_password.dart:
```diff
@@ -0,0 +1,104 @@
+          ),
+          Gap(20.h),
+          AddTextFormField(
+            validator: (value) {
```
can return null

aboelnasrmo replied:
try to use const, you can add it by default when saving

mohamedsyoussef replied:
done

mohamedsyoussef replied:
added return Null

mohamedsyoussef committed:
Add const keyword to LoginBlocListener instantiation

Adding the const keyword to the LoginBlocListener instantiation for improved
performance and efficiency.

mohamedsyoussef committed:
Refactor form validation and navigation in login and onboarding screens

Return null after validating email and password fields for better code readability.
Update navigation method in onboarding screen to use Navigator for consistency.
